tests/aggregation/tests.py

```diff
@@ -2359,3 +2390,256 @@ def test_aggregate_combined_queries(self):
+                delimiter=";",
+                order_by=KeyTextTransform("lang", "json_field"),
+                output_field=CharField(),
+            ),
+        )
+        self.assertEqual(values, {"stringagg": "en;pl"})
+
+    def test_string_agg_filter(self):
+        values = StringAggTestModel.objects.aggregate(
+            stringagg=StringAgg(
+                "char_field",
+                delimiter=";",
+                filter=Q(char_field__endswith="3") | Q(char_field__endswith="1"),
+            )
+        )
+        self.assertEqual(values, {"stringagg": "Foo1;Foo3"})
+
+    @skipUnlessDBFeature("supports_aggregate_order_by_clause")
+    def test_order_by_in_subquery(self):
+        stats = []
+        for i, agg in enumerate(StringAggTestModel.objects.order_by("char_field")):
+            stats.append(StatTestModel(related_field=agg, int1=i, int2=i + 1))
+            stats.append(StatTestModel(related_field=agg, int1=i + 1, int2=i))
+        StatTestModel.objects.bulk_create(stats)
+
+        aggregate = StringAgg(
+            Cast("stattestmodel__int1", CharField()),
+            delimiter=";",
+            order_by="-stattestmodel__int2",
+        )
+        expected_result = [("Foo1", "0;1"), ("Foo2", "1;2"), ("Foo3", "2;3"), ("Foo4", "3;4")]
+
+        subquery = (
+            StringAggTestModel.objects.filter(
+                pk=OuterRef("pk"),
+            )
+            .annotate(agg=aggregate)
+            .values("agg")
+        )
+        values = (
+            StringAggTestModel.objects.annotate(
+                agg=Subquery(subquery),
+            )
+            .order_by("char_field")
+            .values_list("char_field", "agg")
+        )
+        self.assertEqual(list(values), expected_result)
+
+    def test_filter_in_subquery(self):
+        StatTestModel.objects.bulk_create(
+            [
+                StatTestModel(related_field=self.aggs[0], int1=0, int2=5),
+                StatTestModel(related_field=self.aggs[0], int1=1, int2=4),
+                StatTestModel(related_field=self.aggs[0], int1=2, int2=3),
+            ]
+        )
+
+        aggregate = StringAgg(
+            Cast("stattestmodel__int2", CharField()),
+            delimiter=";",
+            filter=Q(stattestmodel__int1__lt=2),
+        )
+        expected_result = [("Foo1", "5;4"), ("Foo2", None)]
+
+        subquery = (
+            StringAggTestModel.objects.filter(
+                pk=OuterRef("pk"),
+            )
+            .annotate(agg=aggregate)
+            .values("agg")
+        )
+        values = (
+            StringAggTestModel.objects.annotate(
+                agg=Subquery(subquery),
+            )
+            .filter(
+                char_field__in=["Foo1", "Foo2"],
+            )
+            .order_by("char_field")
+            .values_list("char_field", "agg")
+        )
+        self.assertEqual(list(values), expected_result)
+
+    def test_filter_in_subquery_with_exclude(self):
```

Comment:
I appreciate the testing but does this cover any particular edge case in subqueries?